eProctoring/assets/admin/js/urlify.js

```
@@ -146,24 +146,51 @@
+            // Use a safer regex for Unicode handling
+            s = XRegExp.replace(s, XRegExp('[^-_\\p{L}\\p{N}\\s]+', 'g'), '');
         } else {
-            s = s.replace(/[^-\w\s]/g, ''); // remove unneeded chars
+            // Remove unneeded characters
+            s = s.replace(/[^-\w\s]+/g, '');
         }
-        s = s.replace(/^\s+|\s+$/g, ''); // trim leading/trailing spaces
-        s = s.replace(/[-\s]+/g, '-'); // convert spaces to hyphens
-        s = s.substring(0, num_chars); // trim to first num_chars chars
-        return s.replace(/-+$/g, ''); // trim any trailing hyphens
+    
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)